ni_measurementlink_service/_internal/parameter/serializer.py

```diff
@@ -158,9 +158,10 @@ def _get_overlapping_parameters(
             raise Exception(
                 f"Error occurred while reading the parameter - given protobuf index '{field_index}' is invalid."
             )
-        type = parameter_metadata_dict[field_index].type
-        is_repeated = parameter_metadata_dict[field_index].repeated
-        decoder = serialization_strategy.get_decoder(type, is_repeated)
+        field_metadata = parameter_metadata_dict[field_index]
+        type = field_metadata.type
+        is_repeated = field_metadata.repeated
+        decoder = serialization_strategy.get_decoder(type, is_repeated, field_metadata.message_type)
```

Comment:
We don't use these again, do we?
```suggestion
        field_metadata = parameter_metadata_dict[field_index]
        decoder = serialization_strategy.get_decoder(field_metadata.type, field_metadata.repeated, field_metadata.message_type)
```